src/main/scala/com/yotpo/metorikku/input/readers/kafka/KafkaInput.scala

```diff
@@ -8,24 +8,38 @@ import org.apache.kafka.clients.consumer.KafkaConsumer
 import org.apache.spark.sql.{DataFrame, SparkSession}
 
 
-case class KafkaInput(name: String, servers: Seq[String], topic: String, consumerGroup: Option[String],
-                      options: Option[Map[String, String]], schemaRegistryUrl: Option[String], schemaSubject:  Option[String]) extends Reader {
+case class KafkaInput(name: String, servers: Seq[String], topic: Option[String], topicPattern: Option[String], consumerGroup: Option[String],
+                      options: Option[Map[String, String]], schemaRegistryUrl: Option[String], schemaSubject: Option[String]) extends Reader {
   @transient lazy val log = org.apache.log4j.LogManager.getLogger(this.getClass)
 
+
   def read(sparkSession: SparkSession): DataFrame = {
     consumerGroup match {
       case Some(group) =>
         log.info(s"creating consumer group with id $group")
         val consumer = createKafkaConsumer(group)
-        val lagWriter = new KafkaLagWriter(consumer, topic)
+        val chosen_topic = topic.getOrElse(topicPattern.getOrElse(""))
```

Comment:
You can't create this here when using topic pattern... you don't know the topics... I'm not even sure you need it in the lag writer